projects/fsastorefrontlib/src/core/my-account/facade/consent.service.ts

```diff
@@ -0,0 +1,38 @@
+import { Injectable } from '@angular/core';
+import { take } from 'rxjs/operators';
+import * as fromAction from './../store/actions';
+import { StateWithMyAccount } from '../store/my-account-state';
+import { select, Store } from '@ngrx/store';
+import * as fromConsentStore from './../store';
+import { UserIdService } from '@spartacus/core';
+import { Observable } from 'rxjs/internal/Observable';
+
+@Injectable()
+export class ConsentService {
+  constructor(
+    protected store: Store<StateWithMyAccount>,
+    protected userIdService: UserIdService
+  ) {}
+
+  loadConsents() {
+    this.userIdService
+      .getUserId()
+      .pipe(take(1))
+      .subscribe(occUserId => {
```

Comment:
You can move this `subscribe` in the component and extend this method with the parameter `userId` to avoid subscription in the service.

Reply:
Done!